pika/connection.py

```diff
@@ -1025,35 +1077,30 @@ def _client_properties(self):
         }
 
     def _close_channels(self, reply_code, reply_text):
-        """Close the open channels with the specified reply_code and reply_text.
+        """Initiate graceful closing of channels that are in OPEN or OPENING
+        states, passing reply_code and reply_text.
 
         :param int reply_code: The code for why the channels are being closed
         :param str reply_text: The text reason for why the channels are closing
 
         """
-        if self.is_open:
-            for channel_number in dictkeys(self._channels):
-                if self._channels[channel_number].is_open:
-                    self._channels[channel_number].close(reply_code, reply_text)
-                else:
-                    del self._channels[channel_number]
-                    # Force any lingering callbacks to be removed
-                    # moved inside else block since channel's _cleanup removes
-                    # callbacks
-                    self.callbacks.cleanup(channel_number)
-        else:
-            self._channels = dict()
+        assert self.is_open, str(self)
 
-    def _combine(self, a, b):
+        for channel_number in dictkeys(self._channels):
```

Comment:
REVIEWME Now also initiates closing of channels that are in OPENING state (in addition to those in OPEN state)